src/rgw/rgw_lua_utils.cc

```diff
@@ -47,29 +31,12 @@ void create_debug_action(lua_State* L, CephContext* cct) {
 }
 
 void stack_dump(lua_State* L) {
-	auto i = lua_gettop(L);
+  int top = lua_gettop(L);
   std::cout << std::endl << " ----------------  Stack Dump ----------------" << std::endl;
-  std::cout << "Stack Size: " << i << std::endl;
-  while (i > 0) {
-    const auto t = lua_type(L, i);
-    switch (t) {
-      case LUA_TNIL:
-        std::cout << i << ": nil" << std::endl;
-        break;
-      case LUA_TSTRING:
-        std::cout << i << ": " << lua_tostring(L, i) << std::endl;
-        break;
-      case LUA_TBOOLEAN:
-        std::cout << i << ": " << lua_toboolean(L, i) << std::endl;
-        break;
-      case LUA_TNUMBER:
-        std::cout << i << ": " << lua_tonumber(L, i) << std::endl;
-        break;
-      default: 
-        std::cout << i << ": " << lua_typename(L, t) << std::endl;
-        break;
-    }
-    i--;
+  std::cout << "Stack Size: " << top << std::endl;
+  for (int i = 1, j = -1; i <= top; i++, j--) {
```

Comment:
j should start from `-top` and then go all the way up to -1. e.g. currently we have a stack printed as:
```
 ----------------  Stack Dump ----------------
Stack Size: 5
[1,-1]: table: 0x56183b1c0680
[2,-2]: table: 0x56183b1c0720
[3,-3]: __index
[4,-4]: userdata: 0x7ffc7dfe5950
[5,-5]: userdata: 0x561839a720bc
```

but should be:
```
 ----------------  Stack Dump ----------------
Stack Size: 5
[1,-5]: table: 0x56183b1c0680
[2,-4]: table: 0x56183b1c0720
[3,-3]: __index
[4,-2]: userdata: 0x7ffc7dfe5950
[5,-1]: userdata: 0x561839a720bc
```

Reply:
Brain fart, thanks.